extension/indexer/external_subscriber_client.go

```diff
@@ -0,0 +1,44 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package indexer
```

Comment:
> Can we move both the client and server into a sub-directory ie. grpcindexer ?
> 
> We should implement the server in a way that is VM agnostic rather than implementing it specifically for MorpheusVM

I've moved `external_subscriber_client.go` into it's own directory (`grpcindexer`). However, I haven't done the same yet for sidecar because of it's dependency on the MorpheusVM parser:

https://github.com/ava-labs/hypersdk/blob/4a6cafc648284c95fd186f3535638ceb3fd64208/examples/morpheusvm/rpc/jsonrpc_client.go#L130C1-L134C2

I think that for the grpc server to be fully VM agnostic, we should move this parser logic outside of MorpheusVM so that the server can utilize the VM-agnostic parser and thus be fully VM-agnostic

Reply:
Why can't we pass this in as this interface instead: https://github.com/ava-labs/hypersdk/blob/86d90df4e07287ab2abe8bd4c79b7bc7a9d2a42a/chain/dependencies.go#L30 ?

Reply:
The parser dependency for the external subscriber server is more so because of the parser being sent during initialization. 

https://github.com/RodrigoVillar/hypersdk/blob/fd26c8185c021b391fcbae9efcdb019daf35884a/examples/morpheusvm/externalsubscriber/sidecar.go#L79-L86

In the case above, we are sending the parser details from the VM to the server, who will then call `rpc.NewParser()` with the values it extracted from the gRPC call. If we generalize the parser, then we can get rid of the Morpheus parser dependency and then be able to move the external subscriber server to `grpcindexer`